[0.36.0.beta.2] Could not use google-plus-ios-sdk - Undefined symbols for architecture

After updating to Cocoapods 0.36 beta 2 I experienced following problems in my Swift project:

I am not able to import `google-plus-ios-sdk` the old way, in my bridging header file:

    #import <GooglePlus/GooglePlus.h> --> FILE NOT FOUND
    #import <GoogleOpenSource/GoogleOpenSource.h> --> FILE NOT FOUND

Which seems to be ok, as for other pods (having the same issue when imported in bridging header) I can simply import proper frameworks in my swift source code files. 
When trying to import Google frameworks in swift it works like expected:

    import google_plus_ios_sdk --> OK, no error here

...but it gives me an compilation error when trying to use any of Google's SDK class:

    Undefined symbols for architecture i386:
    "_OBJC_CLASS_$_GPPURLHandler", referenced from:
      __TMaCSo13GPPURLHandler in AppDelegate.o
    ld: symbol(s) not found for architecture i386
    clang: error: linker command failed with exit code 1 (use -v to see invocation) 

I am able to integrate most of my old, Obj-C pods without an issue, but this one is causing problems like described above. I have already cleaned up project, as well as removed all files stored in Xcode's DeliveredData directory but it didn't solve the problems.

Is there something I am doing wrong, or it's the problem inside the `google-plus-ios-sdk` pod? If so, are there any steps I could make to fix it?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -33,6 +33,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Samuel Giddins](https://github.com/segiddins)
   [#3101](https://github.com/CocoaPods/CocoaPods/issues/3101)
 
+* Do not create pod target if `source_files` only contains headers.  
+  [Boris Bügling](https://github.com/neonichu)
+  [#3106](https://github.com/CocoaPods/CocoaPods/issues/3106)
+
 
 ## 0.37.0.beta.1
 

--- a/lib/cocoapods/target/pod_target.rb
+++ b/lib/cocoapods/target/pod_target.rb
@@ -52,7 +52,9 @@ module Pod
     # A target should not be build if it has no source files.
     #
     def should_build?
-      !file_accessors.flat_map(&:source_files).empty?
+      source_files = file_accessors.flat_map(&:source_files)
+      source_files -= file_accessors.flat_map(&:headers)
+      !source_files.empty?
     end
 
     # @return [Array<Specification::Consumer>] the specification consumers for

--- a/spec/unit/target/pod_target_spec.rb
+++ b/spec/unit/target/pod_target_spec.rb
@@ -63,6 +63,22 @@ module Pod
         end.message
         message.should.match /subspecs across different build configurations/
       end
+
+      it 'builds a pod target if there are actual source files' do
+        fa = Sandbox::FileAccessor.new(nil, @pod_target)
+        fa.stubs(:source_files).returns([Pathname.new('foo.m')])
+        @pod_target.stubs(:file_accessors).returns([fa])
+
+        @pod_target.should_build?.should == true
+      end
+
+      it 'does not build a pod target if there are only header files' do
+        fa = Sandbox::FileAccessor.new(nil, @pod_target)
+        fa.stubs(:source_files).returns([Pathname.new('foo.h')])
+        @pod_target.stubs(:file_accessors).returns([fa])
+
+        @pod_target.should_build?.should == false
+      end
     end
 
     describe 'Support files' do

